compiler/rustc_hir_typeck/src/expr_use_visitor.rs

```diff
@@ -734,7 +734,10 @@ impl<'tcx, Cx: TypeInformationCtxt<'tcx>, D: Delegate<'tcx>> ExprUseVisitor<'tcx
                 // struct; however, when EUV is run during typeck, it
                 // may not. This will generate an error earlier in typeck,
                 // so we can just ignore it.
-                span_bug!(with_expr.span, "with expression doesn't evaluate to a struct");
+                self.cx.tcx().dcx().span_delayed_bug(
```

Comment:
Since an error must have been reported in this typeck, you can just assert that `infcx.tainted_by_errors().is_some()`. That's the nonglobal version of what it was before my PR

Reply:
Done